exec/cnex/support.c

```diff
@@ -0,0 +1,212 @@
+    }
+
+    // Jump to the end of the list; to add this path onto our search paths.
+    while (p && p->next) {
+        p = p->next;
+    }
+
+    // Add this new path to the paths.
+    p->next = n;
+}
+
+void path_freePaths(TPath *paths)
+{
+    TPath *n = paths->next;
+    while (n) {
+        free(n->path);
+        TPath *t = n->next;
+        free(n);
+        n = t;
+    }
+    free(paths->path);
+    free(paths);
+}
+
+// Implementation of POSIC getline() function for Windows
+#ifdef _WIN32
+#define MAX_COUNT   261 /* Max buffer size to read per-pass. */
+static int getline(char **buf, size_t *len, FILE *f)
+{
+    int pos;
+    int c;
+
+    if (buf == NULL || f == NULL || len == NULL) {
+        return -1;
+    }
+
+    c = fgetc(f);
+    if (feof(f)) {
+        return -1;
+    }
+
+    if (*buf == NULL) {
+        *buf = malloc(MAX_COUNT);
+        if (*buf == NULL) {
+            exec_error("Could not allocate %d bytes for path in getline() function.", MAX_COUNT);
+            return -1;
+        }
+        *len = MAX_COUNT;
+    }
+
+    pos = 0;
+    while (!feof(f)) {
+        if (pos + 1 >= *len) {
+            // Double our buffer size, since MAX_COUNT wasn't enough on the last read.
+            size_t size = *len *= 2;
+            if (size < MAX_COUNT) {
+                size = MAX_COUNT;
+            }
+            char *new_ptr = realloc(*buf, size);
+            if (new_ptr == NULL) {
+                exec_error("Could not reallocate %d bytes for path in getline() function.", size);
+                return -1;
+            }
+            *len = size;
+            *buf = new_ptr;
+        }
+        if (c == '\n') {
+            break;
+        }
+        (*buf)[pos++] = (char)c;
+        c = fgetc(f);
+    }
+    (*buf)[pos] = '\0';
+    return pos;
+}
+#endif
+
+void path_initPaths(const char *source_path)
+{
+    // Paths are arlready initialized, so just return.
+    if (neon_paths) {
+        return;
+    }
+
+    // Push neon executable path first.
+    if (strlen(source_path)) {
+        path_pushPath(source_path);
+    }
+    // Then push the cwd path.
+    path_pushPath("./");
+
+    // Get paths from %NEONPATH% environment variable, if there is one.
+    char *ne = getenv("NEONPATH");
+    if (ne != NULL) {
+        char *next;
+        char *path = ne;
+        for (;;) {
+            next = strchr(path, PATHSEP);
+            if (next == NULL) {
+                break;
+            }
+            *next = '\0';
+            path_pushPath(path);
+            path = next + 1;
+        }
+        // Push the last (or perhaps only) path in the list to the stack.
+        if (path != NULL) {
+            path_pushPath(path);
+        }
+    }
+
+    // Next, check for .neonpath, and process any paths that might exist in it.
+    FILE *fp = fopen(".neonpath", "r");
+    if (fp) {
+        char *path = NULL;
+        size_t len = 0;
+        for (;;) {
+            if (getline(&path, &len, fp) == -1) {
```

Comment:
It's completely unnecessary to use getline for this. Use `fgets`.

Reply:
Done.